src/core_atmosphere/dynamics/mpas_atm_time_integration.F

```diff
@@ -6338,36 +6339,123 @@ subroutine atm_rk_dynamics_substep_finish( state, diag, dynamics_substep, dynami
       call mpas_pool_get_array(state, 'rho_zz', rho_zz_1, 1)
       call mpas_pool_get_array(state, 'rho_zz', rho_zz_2, 2)
 
+      MPAS_ACC_TIMER_START('atm_rk_dynamics_substep_finish [ACC_data_xfer]')
+      !$acc enter data create(ru_save, u_1, rtheta_p_save, theta_m_1, rho_p_save, rw_save, &
+      !$acc                   w_1, rho_zz_1) &
+      !$acc            copyin(ru, u_2, rtheta_p, rho_p, theta_m_2, rho_zz_2, rw, &
+      !$acc                   w_2, ruAvg, wwAvg, ruAvg_split, wwAvg_split, rho_zz_old_split)
+      MPAS_ACC_TIMER_STOP('atm_rk_dynamics_substep_finish [ACC_data_xfer]')
+
+      !$acc kernels
+      theta_m_1(:,cellEnd+1) = 0.0_RKIND
+      !$acc end kernels
+
       inv_dynamics_split = 1.0_RKIND / real(dynamics_split)
       
       if (dynamics_substep < dynamics_split) then
+         !$acc parallel
```

Comment:
Should we add `default(present)` clauses here and also to other `!$acc parallel` directives?

Reply:
Yeah makes sense. And fixed this in the update.